### ISSUE TITLE
Envoy endpoints are being removed indefinitely due to node failures

### ISSUE THREAD

**Issue by canthefason** [istio/istio#22102]
**Bug description**
When an Istio-enabled pod gets rescheduled on a new node due to node failure, the Envoy endpoint is being removed indefinitely even though the pod gets back into healthy state on the new node. I specifically checked the Envoy health by making a curl request to localhost:15000/ready for the pod in question and it returns a 200 response. This pod is also being listed with no issues when I call `istioctl ps`

Another interesting finding is, for the last node failure we faced, six of the Istio-enabled pods got rescheduled, and only the endpoints for two of the pods were missing. The other four were working as expected. And one other finding is, those two of the failing pods started running on the same node after the rescheduling which might give us a clue. 

In addition to this, these pods are being listed in the Pilot logs under `pilot_eds_no_instances` push status. 

**Expected behavior**
After a pod rescheduling I would eventually expect all Istio-enabled pods to switch into healthy state. 
 
**Steps to reproduce the bug**
In our case, we've identified that node failures were causing this. However, considering that not all the rescheduled pods are being impacted by this, it might be some sort of edge case or a race condition. 

**Version (include the output of `istioctl version --remote` and `kubectl version` and `helm version` if you used Helm)**

```
$ istioctl version
client version: 1.4.3
control plane version: 1.4.3
data plane version: 1.4.3 (10 proxies)
```

```
Client Version: version.Info{Major:"1", Minor:"17", GitVersion:"v1.17.3", GitCommit:"06ad960bfd03b39c8310aaf92d1e7c12ce618213", GitTreeState:"clean", BuildDate:"2020-02-13T18:06:54Z", GoVersion:"go1.13.8", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"15+", GitVersion:"v1.15.9-gke.9", GitCommit:"a9973cbb2722793e2ea08d20880633ca61d3e669", GitTreeState:"clean", BuildDate:"2020-02-07T22:35:02Z", GoVersion:"go1.12.12b4", Compiler:"gc", Platform:"linux/amd64"}
```

**How was Istio installed?**
Istio is installed by running istio/installer via [Isopod](https://github.com/cruise-automation/isopod). 

**Environment where bug was observed (cloud vendor, OS, etc)**
GCP GKE

**Comment by howardjohn** [comment]
might be related to https://github.com/istio/istio/issues/20676, fixed in 1.4.5?

**Comment by canthefason** [comment]
It is highly likely. I've checked the pilot logs regarding [your comment here](https://github.com/istio/istio/issues/20733#issuecomment-582506427), and I was able to confirm that we've received similar logs for those two problematic pods during that time window that node failure happened. I'll soon rollout that newer version and keep you updated about the end result.  

Thanks for your prompt response!
